pkgs/build-support/kernel/modules-closure.sh

```diff
@@ -33,4 +33,13 @@ for module in $closure; do
     echo $target >> $out/insmod-list
 done
 
+mkdir -p $out/lib/firmware
+for module in $closure; do
+    for i in $(modinfo -F firmware $module); do
+        mkdir -p "$out/lib/firmware/$(dirname "$i")"
+        echo "firmware: $i"
```

Comment:
Maybe print the name of the module that needs the firmware as well?

LGTM otherwise.

Reply:
Fixed.